ordered_product/models.py

```diff
@@ -0,0 +1,101 @@
+from product.models import Product
+
+
+class User(models.Model):
+    user_name = models.CharField(max_length=32,
+                                 primary_key=True,
+                                 null=False,
+                                 validators=[MinLengthValidator(6)],
+                                 help_text="User Name")
+    first_name = models.CharField(max_length=32,
+                                  validators=[MinLengthValidator(1)],
+                                  help_text="First Name")
+    last_name = models.CharField(max_length=32,
+                                 validators=[MinLengthValidator(1)],
+                                 help_text="Last Name")
+    password = models.CharField(max_length=16,
+                                validators=[MinLengthValidator(6)],
+                                null=False,
+                                help_text="Password")
+    user_type = models.CharField(max_length=16,
+                                 validators=[MinLengthValidator(6)],
+                                 null=False,
+                                 choices=[('client', 'client'), ('supplier', 'supplier')],
+                                 help_text="User Type - supplier/client")
+
+    class Meta:
+        abstract = True
+
+
+class Supplier(User):
+    business_name = models.CharField(max_length=32,
+                                     validators=[MinLengthValidator(6)],
+                                     null=False,
+                                     help_text="Business Name")
+
+
+class DeliveryLocation(models.Model):
+    user_name = models.ForeignKey(Supplier, on_delete=models.CASCADE)
+    location = models.CharField(max_length=32, validators=[MinLengthValidator(1)])
+    date = models.DateField()
+
+
+class Client(User):
+    area = models.CharField(max_length=32,
+                            validators=[MinLengthValidator(6)],
+                            help_text="area")
+
+
+class SupplierProduct(models.Model):
+    suppler_product_id = models.IntegerField(primary_key=True)
+    qr_code = models.ForeignKey(Product, on_delete=models.CASCADE)
+    user_name = models.ForeignKey(Supplier, on_delete=models.CASCADE)
+    price = models.IntegerField(validators=[MinValueValidator(1)])
+    quantity = models.IntegerField(validators=[MinValueValidator(1)])
+
+    class Meta:
+        unique_together = ('suppler_product_id', 'qr_code')
+
+
+class OrderdProduct(models.Model):
+    ordered_product_id = models.IntegerField(primary_key=True, editable=False, unique=True)
+    delivery_location_id = models.ForeignKey(DeliveryLocation, on_delete=models.CASCADE)
+    user_name = models.ForeignKey(Client, on_delete=models.CASCADE)
+    supplier_product_id = models.ForeignKey(SupplierProduct, on_delete=models.CASCADE)
+    quantity = models.IntegerField(default=1, validators=[MinValueValidator(1)])
+
+    @staticmethod
+    def filter_ordered_prodct_ID(opID):
```

Comment:
change variable and function name to snake_case with lowercase only. Also for the rest of the PR.